xdsl/dialects/riscv.py

```diff
@@ -729,20 +707,20 @@ class RdRsImmIntegerOperation(IRDLOperation, RISCVInstruction, ABC):
     This is called I-Type in the RISC-V specification.
     """
 
-    rd: OpResult = result_def(IntRegisterType)
-    rs1: Operand = operand_def(IntRegisterType)
-    immediate: AnyIntegerAttr | LabelAttr = attr_def(AnyIntegerAttr | LabelAttr)
+    rd = result_def(IntRegisterType)
+    rs1 = operand_def(IntRegisterType)
+    immediate = attr_def(IntegerAttr[IntegerType] | LabelAttr)
```

Comment:
I just realised that this doesn't actually check the bitwidth, I'll leave it as-is and try and fix in a follow-up PR.